extension/intents/tabs/tabs.js

```diff
@@ -372,6 +372,178 @@ intentRunner.registerIntent({
   },
 });
 
+async function collectTabsAround(center, tabs) {
+  const tabsLeft = tabs.filter(tab => {
+    return tab.index < center.index;
+  });
+
+  const tabsRight = tabs.filter(tab => {
+    return tab.index > center.index;
+  });
+
+  for (let i = 0; i < tabsLeft.length; i++) {
+    await browser.tabs.move(tabsLeft[tabsLeft.length - i - 1].id, {
+      index: center.index - i - 1,
+    });
+  }
+
+  for (let i = 0; i < tabsRight.length; i++) {
+    await browser.tabs.move(tabsRight[i].id, { index: center.index + i + 1 });
+  }
+}
+
+async function collectRightTabs(tabs) {
+  const leftestTab = tabs[0];
+  const tabsRight = tabs.slice(1);
+
+  for (let i = 0; i < tabsRight.length; i++) {
+    await browser.tabs.move(tabsRight[i].id, {
+      index: leftestTab.index + i + 1,
+    });
+  }
+}
+
+function groupTabs(tabs, keyGetter) {
+  const groupedTabs = {};
+
+  for (let i = 0; i < tabs.length; i++) {
+    const key = keyGetter(tabs[i]);
+
+    if (groupedTabs[key] === undefined) {
+      groupedTabs[key] = [tabs[i]];
+    } else {
+      groupedTabs[key].push(tabs[i]);
+    }
+  }
+
+  return groupedTabs;
+}
+
+async function getMatchingTabs(options) {
+  const tabQuery = {
+    pinned: false,
+    windowType: "normal",
+  };
+
+  if (options.allWindows === false) {
+    tabQuery.currentWindow = true;
+  }
+
+  if (options.query === undefined && options.activeTab !== undefined) {
+    const activeTab = options.activeTab;
+    const activeHostname = new URL(activeTab.url).hostname;
+    tabQuery.url = "*://*." + activeHostname + "/*";
+  }
+
+  let matchingTabs = await browser.tabs.query(tabQuery);
+
+  if (options.query !== undefined) {
+    matchingTabs = matchingTabs.filter(tab => {
+      return (
+        new URL(tab.url).origin.includes(options.query) ||
+        tab.title.toLowerCase().includes(options.query)
```

Comment:
The query can also be upper case, so you should also make sure to convert it to lower case as well.